src/text/text.c

```diff
@@ -76,11 +84,39 @@ static n_GPoint n_graphics_prv_draw_text_line(n_GContext * ctx, const char * tex
     return text_origin;
 }
 
+static n_GPoint n_graphics_prv_get_aligned_text_origin(const char * text, 
+    uint32_t idx, uint32_t indx_end, const n_GRect box, const n_GPoint line_origin,
+    int8_t * glyph_widths, uint8_t padding, const n_GTextAlignment alignment) {
+
+    // Calculate origin of the character
+    // where we are breaking the line.
+    n_GPoint breakable_char_origin = box.origin;
+    for (int i = idx; i < indx_end; i++)
```

Comment:
Does this break for multi-byte glyphs?

Reply:
I think you might have to pull in the `next_codepoint` / `next_index` logic from inside `draw_text_ex`.

Reply:
Good point. I'll add that some multi-byte codepoints to the test app fix up as needed.